receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -116,17 +117,45 @@ func (kr *k8sobjectsreceiver) Start(ctx context.Context, _ component.Host) error
 	}
 
 	if len(validConfigs) == 0 {
-		err := errors.New("no valid Kubernetes objects found to watch")
+		err = errors.New("no valid Kubernetes objects found to watch")
 		return err
 	}
 
-	kr.setting.Logger.Info("Object Receiver started")
 	cctx, cancel := context.WithCancel(ctx)
 	kr.cancel = cancel
 
-	for _, object := range validConfigs {
-		kr.start(cctx, object)
+	if kr.config.K8sLeaderElector != nil {
+		k8sLeaderElector := host.GetExtensions()[*kr.config.K8sLeaderElector]
+		if k8sLeaderElector == nil {
+			return fmt.Errorf("unknown k8s leader elector %q", kr.config.K8sLeaderElector)
+		}
+
+		kr.setting.Logger.Info("trying to become the leader")
```

Comment:
The receiver only register itself at this point, it doesn't try to become the leader. This should be something like "registering the receiver in leader election"